my_project/tab_select/app_select.py

```diff
@@ -142,9 +142,7 @@ def submitted_data(
                 messages_alert["not_available"],
                 "warning",
             )
-        df, location_info = create_df(
-            lines, url_store
-        )  # we might need to split this call into two, one returns df and one returns location_info
+        location_info = get_location_info(lines, url_store)  # we might need to split this call into two, one returns df and one returns location_info
         return (
```

Comment:
Since the df-store is output in the switch_si_ip() function, here, it is no need to get the df info, so I created a new function named 'get_location_info()' . In this way, the submit_data() function will only get the lines and the location info, and the switch_si_ip() will get the df info. create_df() can only be called once.